torch/serialization.py

```diff
@@ -453,14 +453,19 @@ def persistent_load(saved_id):
             raise RuntimeError("Unknown saved id type: %s" % saved_id[0])
 
     f_is_real_file = _is_real_file(f)
-    if f_is_real_file and f.tell() == 0:
-        # legacy_load requires that f has fileno()
-        # only if offset is zero we can attempt the legacy tar file loader
-        try:
-            return legacy_load(f)
-        except tarfile.TarError:
-            # if not a tarfile, reset file offset and proceed
-            f.seek(0)
+
+    try:
+        if f_is_real_file and f.tell() == 0:
+            # legacy_load requires that f has fileno()
+            # only if offset is zero we can attempt the legacy tar file loader
+            try:
+                return legacy_load(f)
+            except tarfile.TarError:
+                # if not a tarfile, reset file offset and proceed
+                f.seek(0)
+    except (io.UnsupportedOperation, AttributeError):
+        raise RuntimeError("You can only torch.load from a file that is seekable. " +
```

Comment:
You don't know that `UnsupportedOperation` has been raised from `seek` if you write it out this way

Reply:
Great point! I will fix this.